netbox/dcim/forms.py

```diff
@@ -2826,6 +2852,10 @@ def __init__(self, *args, **kwargs):
                 device__in=[device, device.get_vc_master()],
                 type=InterfaceTypeChoices.TYPE_LAG
             )
+
+            # Add current site to VLANs query params
+            self.fields['untagged_vlan'].widget.add_additional_query_param('site_id', device.site.pk)
```

Comment:
This approach doesn't seem very DRY: We're setting additional query parameters in two places, per field, per form. It's inevitable that we'll missing some of these references should the parameters need to be changed in the future.

Additionally, these fields need to be updated for the interface forms in `virtualization.forms` as well.

Reply:
That's a valid concern, and – as you pointed out – virtualization is clear example of this inconsistency. #4057 should improve the implementation of this; I'm happy to close this PR until that issue is closed; let me know what's your preference on this.

Reply:
Let's leave this PR open for now. I don't have a read yet on just how involved merging those forms will be: If it's too much effort to take on at the moment, we can push this through.

Reply:
Fixed `virtualization.forms`.